### PR TITLE
Add json proto generation to script

### PR DESCRIPTION
**Pull request by tguinot**
Solves first point of #237

### CODE REVIEW

File: script/gettools.sh [hunk 1]
```diff
@@ -6,6 +6,7 @@ gettools_develop() {
 	tools="github.com/bufbuild/buf/cmd/buf@v1.0.0-rc12
 google.golang.org/protobuf/cmd/protoc-gen-go@v1.27.1
 google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2.0
+github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema@latest
```
Comment (jeremyletang) [inline script/gettools.sh]: Is it json schema that is required or openapiv3? Not sure if there's a difference, but pretty sure the initial ticket asked for this?

**Comment by jeremyletang** [comment]
@tguinot not sure this implements what the initial ticket was requesting? Although after a little search I can see that grpc-gateway does not implements open api v3 but only v2. 
This ticket speaks about using some over plugin to generate openapi v3 though: https://github.com/grpc-ecosystem/grpc-gateway/issues/441

**Comment by jeremyletang** [comment]
Alright, I was mistaken, this is the json representation of the protobuf part. @edd have you check into this? Does that fits your needs?

**Comment by pgs31** [comment]
I think the JSON representation will depend a bit on the implementation of the OpenAPIv3 <-> grpc gateway. Like Jeremey said it looks like grpc-gateway doesn't currently support this, are we writing something ourselves to do the same thing or using some other tool. For example, your protoc command thomas:

protoc --jsonschema_out=./generated/json/veg

Doesn't know anything about the mappings we put in 'grpc--rest-bindings.yml'.. so are we writing a whole different HTTP api? I admit I misunderstood the scope of this, I thought it was just a matter of making slightly different JSON files to describe our current API.

**Comment by edd** [comment]
Just to re-add some context:

The documentation at https://docs.vega.xyz generates Protobuf documentation in this script:

https://github.com/vegaprotocol/documentation/blob/bf2f953dd6dd09070f23c44d02710555bfaf30d5/scripts/build.sh#L28-L45

This depends on this file:

https://github.com/vegaprotocol/sdk-docs/blob/develop/generated/doc.json

which is in this repo:

https://github.com/vegaprotocol/sdk-docs/

Which is not being updated automatically. This repository only exists for documentation's sake and was created when the protos weren't in a public repository. Therefore if we can have this same JSON file generated in in the protos repo, we are one step closer to retiring `sdk-docs`.

**Comment by emilbayes** [comment]
I haven't found a way this can be done with the buf version we are currently using. You can setup jsonschema as a plugin, but selecting only specific directories I haven't seen (or found docs for) with that version. In version v1 I believe you can add a nested buf configuration to any directory that should have special treatment

**Comment by edd** [comment]
This should cover how to generate it with `protoc` rather than `buf`:
@emilbayes is https://github.com/protobuffet/docusaurus-protobuffet/tree/master/packages/docusaurus-protobuffet#installation

**Comment by peterbarrow** [comment]
This PR has been replaced by a new one referencing a new code branch due to comment naming issues.

#353

**Pull request closed without merging**